### PR TITLE
Preparation for gofeatures deprecation

### PR DESCRIPTION
**Pull request by unmultimedio**
README and CODEOWNERS for future updates of `protocolbuffers/wellknowntypes` and `protocolbuffers/gofeatures` states.

You can see individual commits for testing.

### CODE REVIEW

File: modules/static/protocolbuffers/gofeatures/buf.md
Comment (jhump) [inline modules/static/protocolbuffers/gofeatures/buf.md:3]: This should specifically mention v29.2 -- if a user is using any older version then they can't use wellknowntypes instead.

Do you know if the Markdown rendering library we use in the BSR supports call-outs? If so, that would be nice, to make the deprecation visually obvious even if someone doesn't read all of the text.

In GitHub-flavored Markdown it would look something like so:
```md
>[!WARNING]
> This module has been **deprecated**. As of v29.2, use the `protocolbuffers/wellknowntypes` module instead.
```

With Docusaurus-style callouts, it might look like this instead:

```md
:::warning

This module has been **deprecated**. As of v29.2, use the `protocolbuffers/wellknowntypes` module instead.

:::
```

File: .github/CODEOWNERS
Comment (jhump) [inline .github/CODEOWNERS:2]: Do we need to do anything else to enable this, like a module setting to actually require approvals from code owners? (I haven't messed around with code owners in quite some time...)

Comment (unmultimedio) [reply .github/CODEOWNERS:2]: Imo with the fact that we're notified that's enough, usually people that approve PRs from this repo are no more than 4-5 people. But if we want to lock it down to _require_ a CODEOWNER approval (if it's not already on), that'd be @rubensf

File: modules/static/protocolbuffers/gofeatures/buf.md
Comment (unmultimedio) [reply modules/static/protocolbuffers/gofeatures/buf.md:3]: > This should specifically mention v29.2 -- if a user is using any older version then they can't use wellknowntypes instead.

Sure, I'll say that.

> Do you know if the Markdown rendering library we use in the BSR supports call-outs? If so, that would be nice, to make the deprecation visually obvious even if someone doesn't read all of the text.

cc @paul-sachs which one should we use for BSR Readmes?

Comment (paul-sachs) [reply modules/static/protocolbuffers/gofeatures/buf.md:3]: I don't think so. We're just using GitHub flavoured markdown and I'm not aware of call-out support within that variant.

File: .github/CODEOWNERS
Comment (rubensf) [reply .github/CODEOWNERS:2]: Just enabled the requirement of code owners.

Note that even if the author is a code owner, one of the reviewers will need to be a _different_ code owner.

File: modules/static/protocolbuffers/gofeatures/buf.md
Comment (unmultimedio) [reply modules/static/protocolbuffers/gofeatures/buf.md:3]: SCR-20241219-mscu

Tried locally, as Paul says I don't think there's a way to do call-outs 😞 so I think just a quote or H2/H3 is good enough?

File: modules/static/protocolbuffers/gofeatures/buf.md
Comment (jhump) [inline modules/static/protocolbuffers/gofeatures/buf.md:4]: I know I suggested this text. But now that I'm reading it, it's not as clear as it could be. My bad 🤦 

It's unclear that v29.2 refers to a version of wellknowntypes, vs. this module. How about this instead:
```suggestion
> This module has been **deprecated**. Use the `protocolbuffers/wellknowntypes` module, v29.2 or higher,
> instead.
```

**Review by jhump — APPROVED** [review]

File: modules/static/protocolbuffers/gofeatures/buf.md
Comment (unmultimedio) [reply modules/static/protocolbuffers/gofeatures/buf.md]: Changed, thanks!

**Pull request merged**
